### PR TITLE
Remove confusing job!="" filteringSelector when no selector is required.

### PR DESCRIPTION
**Pull request by v-zhuravlev**
since filteringSelector is now optional (can be set to ""), 
remove confusing job!="" filteringSelector when no selector is required,
from some mixins/libs:
- windows
- kafka-lib
- zookeeper-lib
- process-lib
- jvm-lib

This should help with mixins rendered on 
monitoring-mixin website:
https://monitoring.mixins.dev/jvm/

### CODE REVIEW

**Comment by v-zhuravlev** [comment]
To pass tests, we need to merge # 1372 separately, then rebase here.

File: common-lib/common/variables/variables.libsonnet [hunk 1]
```diff
@@ -13,6 +13,11 @@ local utils = import '../utils.libsonnet';
     prometheusDatasourceName=if enableLokiLogs then 'prometheus_datasource' else 'datasource',
     prometheusDatasourceLabel=if enableLokiLogs then 'Prometheus data source' else 'Data source',
   ): {
+       // strip trailing or starting comma if present:
+       // while trailing comman is accepted in PromQL expressions, starting comma is not.
```
Comment (Dasomeone) [inline common-lib/common/variables/variables.libsonnet:17]: Typo fix suggestion in dependency PR

**Review by Dasomeone — COMMENTED** [review]
Aside from the comments on the dependency PR this looks good :)

File: common-lib/common/variables/variables.libsonnet (hunk 1 above)
Comment (v-zhuravlev) [reply common-lib/common/variables/variables.libsonnet]: should be fixed by merge.

**Review by Dasomeone — APPROVED** [review]

**Pull request merged**